src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,201 @@
+            } catch (DateTimeParseException e) {
+                throw new InvalidDateTimeException();
+            }
+        }
+    }
+
+    private Command prepareEvent(String[] arguments) throws EmptyArgumentException, InvalidDateTimeException {
+        if (arguments.length == 1) {
+            throw new EmptyArgumentException("Please input a valid task description!");
+        } else {
+            String description = arguments[1];
+            String[] taskInputAndDate = description.split("/", 2);
+
+            taskInputAndDate[0] = taskInputAndDate[0].trim();
+            taskInputAndDate[1] = taskInputAndDate[1].trim();
+
+            try {
+                LocalDateTime dateTime = LocalDateTime.parse(taskInputAndDate[1].substring(3), FORMATTER);
+                return new EventCommand(this.taskList, this.ui, this.storage, taskInputAndDate[0], dateTime);
+            } catch (DateTimeParseException e) {
+                throw new InvalidDateTimeException();
+            }
+        }
+    }
+
+    private Command prepareFind(String[] arguments) throws EmptyArgumentException, EmptyListException {
+        if (arguments.length == 1) {
+            throw new EmptyArgumentException("Please pass a word after the 'find' command!");
+        } else {
+            if (this.taskList.getList().size() == 0) {
+                throw new EmptyListException();
+            } else {
+                return new FindCommand(this.taskList, this.ui, this.storage, arguments[1]);
+            }
+        }
+    }
+
+    private int calcListPos(String taskIndex, String command) throws InvalidIndexInputException {
+        Matcher m = checkNum.matcher(taskIndex);
+        if (m.find()) {
+            return Integer.parseInt(taskIndex) - 1;
+        } else {
+            throw new InvalidIndexInputException("'" + command + "' is command word; please pass a numerical index or "
+                    + "start your task with another word!");
+        }
+    }
+
+    private Command prepareDone(String[] arguments) throws InvalidIndexInputException, EmptyArgumentException {
+        if (arguments.length == 1) {
+            throw new EmptyArgumentException("Please pass an index after the 'done' command!");
+        } else {
+            int position = calcListPos(arguments[1], arguments[0]);
+
+            if (this.taskList.getList().size() == 0) {
+                throw new InvalidIndexInputException("You have already done all tasks!");
+            } else if (position >= this.taskList.getList().size() || position < 0) {
+                throw new InvalidIndexInputException("Please input an index from 1 to "
+                        + this.taskList.getList().size() + "!");
+            } else {
+                return new DoneCommand(this.taskList, this.ui, this.storage, position);
+            }
+        }
+    }
+
+    private Command prepareDelete(String[] arguments) throws InvalidIndexInputException, EmptyArgumentException {
```

Comment:
Even though the line length is lesser than 120 chars, but perhaps it is better to keep it short and neat. Maybe you can break at 'throws InvalidIndexInputException....' to improve the readability of the code. Same for other similar methods!